test/archive.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const path = require('path');
+const test = require('supertape');
+const archive = require('../server/archive');
+
+test('archive: shows zip archive listing', async (t) => {
+    const file = path.resolve('./test/test.zip');
+    t.equal(archive.isArchive(file), true, 'Should be an archive file');
+    try {
+        const resp = await archive.listArchive(file);
+        t.equal(resp, `delete.js
+get.js
+fixture
+fixture/get.json
+fixture/get-raw
+fixture/get-raw/hello.txt
+fixture/get
+fixture/get/hello.txt
+fixture/get-raw.json
+options.js`);
+    } catch(ex) {
+        t.fail(ex);
+    }
+    
+    t.end();
+});
+
+test('archive: returns false for non zip files', async (t) => {
+    const file = path.resolve('./test/not.a.zip.file');
+    t.equal(archive.isArchive(file), false, 'Should not be an archive file');
+    try {
```

Comment:
We can avoid using `try-catch` here if `listArchive` will contain one :).
Please use one assertion for one test.